app/components/autocomplete/slash_suggestion/slash_suggestion.js

```diff
@@ -57,47 +59,71 @@ export default class SlashSuggestion extends PureComponent {
             value: nextValue,
         } = nextProps;
 
-        if (currentTeamId !== nextTeamId) {
-            this.setState({
-                lastCommandRequest: 0,
-            });
-        }
-
-        const match = nextValue.match(SLASH_REGEX);
-
-        if (!match || this.state.suggestionComplete) {
+        if (nextValue[0] !== '/') {
             this.setState({
                 active: false,
-                matchTerm: null,
-                suggestionComplete: false,
             });
             this.props.onResultCountChange(0);
             return;
         }
 
-        const dataIsStale = Date.now() - this.state.lastCommandRequest > TIME_BEFORE_NEXT_COMMAND_REQUEST;
+        if (nextValue.indexOf(' ') === -1) { // return suggestions for a top level cached commands
+            if (currentTeamId !== nextTeamId) {
+                this.setState({
+                    lastCommandRequest: 0,
+                });
+            }
+
+            const dataIsStale = Date.now() - this.state.lastCommandRequest > TIME_BEFORE_NEXT_COMMAND_REQUEST;
+
+            if ((!nextCommands.length || dataIsStale)) {
+                this.props.actions.getAutocompleteCommands(nextProps.currentTeamId);
+                this.setState({
+                    lastCommandRequest: Date.now(),
+                });
+            }
 
-        if ((!nextCommands.length || dataIsStale)) {
-            this.props.actions.getAutocompleteCommands(nextProps.currentTeamId);
+            const matches = this.filterSlashSuggestions(nextValue.substring(1), nextCommands);
+            this.updateSuggestions(matches);
+        } else if (isMinimumServerVersion(Client4.getServerVersion(), 5, 24)) {
+            const args = {
+                channel_id: this.props.channelId,
+                ...(this.props.rootId && {root_id: this.props.rootId, parent_id: this.props.rootId}),
+            };
+
+            Client4.getCommandAutocompleteSuggestionsList(nextProps.value, nextProps.currentTeamId, args).then(
```

Comment:
Why client calls in the component? this should be an action, then you  can use the result returned by the action no problem...

Also you should handle the case when the request fails.